### ISSUE TITLE
dynamic slot bug?

### ISSUE THREAD

**Issue by vb8448** [EmilStenstrom/django-components#763]
version: 0.102

I have the following component:

```
@register("demo")
class Demo(Component):
    """Helper component to render a model detail view."""
    def get_context_data(self, fields: list) -> dict:
        return {
            "fields": fields,
        }

    # language=HTML
    template = """
        <table class="table">
        {% for field in fields %}
            {% slot name="field-{{ field }}" %}
              <tr>
                  <td>{{ field }}</td>
                  <td>{{ field }}</td>
              </tr>
            {% endslot %}
        {% endfor %}
        </table>
    """
```

when i use it like `{% demo fields /%}` (fields is a list of strings) it works.

When i try to override a single slot like this:

```
    {% demo fields %}
      {% fill "field-data" %}
        ciao
      {% endfill %}
    {% enddemo %}
```

i get

```
An error occured while rendering component 'demo':
TemplateSyntaxError("Component 'demo' passed fill that refers to undefined slot: 'field-data'.\nUnfilled slot names are: ['field-'].\nDid you mean 'field-'?")
```

For some reason, it doesn't expand correctly the field in the slot name. Any idea?

Thanks.

**Comment by JuroOravec** [comment]
Hey @vb8448, this is indeed a bug, I came across it recently too. The issue is that currently the variables used with `{% slot %}` and `{% fill %}` are evaluated earlier than they should be. So in your example we the variable `field` in `{% slot %}` is evaluated to empty string, because the variable is not found at the time django-components evaluates it.

This is being fixed in https://github.com/EmilStenstrom/django-components/pull/758, but it's a big refactor, so might take a while a bit to review.

**Comment by vb8448** [comment]
@JuroOravec I've tried the #758 and it fixed the issues. I'll keep using branch and let you know if something else breaks.

Thank you.

**Comment by JuroOravec** [comment]
@vb8448 This has now been released in [v0.110](https://github.com/EmilStenstrom/django-components/releases/tag/0.110) :)
